From e84d8dd13d31de4942bbf4ec8093932961b9d1b9 Mon Sep 17 00:00:00 2001
From: Xiang Xiao <xiaoxiang@xiaomi.com>
Date: Tue, 21 Apr 2020 22:02:38 +0800
Subject: [PATCH] nuttx: change sched_kfree to metal_free_memory

since the upstream remove sched_kfree and put the similar logic into kmm_free:
https://github.com/apache/incubator-nuttx/commit/231ad202ee0a3abfd2e14fab1f218b588f61284b

Signed-off-by: Xiang Xiao <xiaoxiang@xiaomi.com>
---
 lib/system/nuttx/irq.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/system/nuttx/irq.c libmetal/lib/system/nuttx/irq.c
index 476eab9..84200d3 100644
--- a/lib/system/nuttx/irq.c
+++ libmetal/lib/system/nuttx/irq.c
@@ -45,7 +45,7 @@ static int metal_cntr_irq_handler(int irq, void *context, void *data)
 
 	/* context == NULL mean unregister */
 	irqchain_detach(irq, metal_cntr_irq_handler, data);
-	sched_kfree(data);
+	metal_free_memory(data);
 	return 0;
 }
 
-- 
2.17.1

